pkg/cmd/cli/cmd/expose.go

```diff
@@ -56,7 +56,13 @@ func NewCmdExpose(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.C
 		cmdutil.CheckErr(err)
 		defRun(cmd, args)
 	}
-	cmd.Flags().String("hostname", "", "Set a hostname for the new route")
+	cmd.Flags().String("hostname", "", "Set a hostname for the new route. Optional.")
+	cmd.Flags().String("path", "", "Path that the router watches to route traffic to the service. Optional.")
+	cmd.Flags().String("tls-termination", "", "Indicate a termination type for the generated route. If not set, any termination config will be ignored. Supports 'edge', 'passthrough', or 'reencrypt'.")
+	cmd.Flags().String("cert-file", "", "Provide certificate contents for a secure route. Used by edge and re-encryption terminated routes.")
```

Comment:
The part about meaningless flags is covered by https://github.com/openshift/origin/pull/5968.

Regarding implementing this upstream for ingress, think about it as a carry. If it doesn't match, we have to conform. So in our case I guess we would have to support two different flags for each option (supposing the upstream names will be different). Still not something that is caused by generators and they still won't stand in your way.